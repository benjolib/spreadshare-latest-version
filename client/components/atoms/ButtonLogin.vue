<template>
  <nuxt-link
    :class="$style.login"
    to="/login"
  >
    <span>
      Join Us
      <span :class="$style.viaText">
        via
      </span>
    </span>
    <fa
      :class="$style.icon"
      :icon="faFacebook"
      size="xs"
    />
    <fa
      :class="$style.icon"
      :icon="faTwitter"
      size="sm"
    />
    <fa
      :class="$style.icon"
      :icon="faGoogle"
      size="xs"
    />
  </nuxt-link>
</template>

<script>
import {
  faFacebookF,
  faTwitter,
  faGoogle,
} from '@fortawesome/free-brands-svg-icons'
export default {
  computed: {
    faFacebook: () => faFacebookF,
    faTwitter: () => faTwitter,
    faGoogle: () => faGoogle,
  },
}
</script>

<style lang="scss" module>
@import '@design';

.login {
  @extend %button;

  display: inline-flex;
  width: 110px;
  height: 48px;
  color: #fff !important;
  background: #2b60f5;
  border: none;
  box-shadow: none;

  @include mq(large-and-xlarge-devices) {
    width: 200px;
  }
}

.viaText {
  display: none;
  @include mq(large-and-xlarge-devices) {
    display: inline;
    margin-right: 8px;
    font-style: italic;
    font-weight: 100;
  }
}

.icon {
  display: none;
  margin: 0 4px;
  @include mq(large-and-xlarge-devices) {
    display: inline;
  }
}
</style>
