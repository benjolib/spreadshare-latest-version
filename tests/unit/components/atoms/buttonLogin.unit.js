// @Flow

import { shallowMount } from '@vue/test-utils'
import ButtonLogin from '@atoms/ButtonLogin'

describe('buttonLogin', () => {
  const wrapper = shallowMount(ButtonLogin, {
    stubs: ['nuxt-link', 'fa'],
  })

  describe('unit tests', () => {
    test('is a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper.is(ButtonLogin)).toBeTruthy()
    })

    test('should have a nuxt-link pointing to /login', () => {
      const link = wrapper.find('nuxt-link-stub')
      expect(link.attributes().to).toBe('/login')
    })

    test('link should have a login class', () => {
      const link = wrapper.find('nuxt-link-stub')
      expect(link.classes()).toContain('login')
    })

    test('link should have proper text', () => {
      const spans = wrapper.findAll('span')

      const spanOuter = spans.at(0)
      expect(spanOuter.text()).toContain('Join Us')

      const spanInner = spans.at(1)
      expect(spanInner.text()).toContain('via')
      expect(spanInner.classes()).toContain('viaText')
    })

    test('should have a facebook icon', () => {
      const cmp = shallowMount(ButtonLogin, {
        stubs: ['nuxt-link', 'fa'],
        computed: {
          faFacebook: () => 'fbIcon',
        },
      })
      const icons = cmp.findAll('fa-stub')
      const fbIcon = icons.at(0)
      expect(fbIcon.attributes().size).toEqual('xs')
      expect(fbIcon.attributes().icon).toEqual('fbIcon')
      expect(fbIcon.classes()).toContain('icon')
    })

    test('should have a twitter icon', () => {
      const cmp = shallowMount(ButtonLogin, {
        stubs: ['nuxt-link', 'fa'],
        computed: {
          faTwitter: () => 'twitterIcon',
        },
      })
      const icons = cmp.findAll('fa-stub')
      const fbIcon = icons.at(1)
      expect(fbIcon.attributes().size).toEqual('sm')
      expect(fbIcon.attributes().icon).toEqual('twitterIcon')
      expect(fbIcon.classes()).toContain('icon')
    })

    test('should have a google icon', () => {
      const cmp = shallowMount(ButtonLogin, {
        stubs: ['nuxt-link', 'fa'],
        computed: {
          faGoogle: () => 'googleIcon',
        },
      })
      const icons = cmp.findAll('fa-stub')
      const fbIcon = icons.at(2)
      expect(fbIcon.attributes().size).toEqual('xs')
      expect(fbIcon.attributes().icon).toEqual('googleIcon')
      expect(fbIcon.classes()).toContain('icon')
    })
  })

  describe('snapshot tests', () => {
    test('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
