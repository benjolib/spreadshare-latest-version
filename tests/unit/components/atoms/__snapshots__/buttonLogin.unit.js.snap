// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonLogin snapshot tests should match the snapshot 1`] = `
<nuxt-link-stub
  class="login"
  to="/login"
>
  <span>
    
    Join Us
    
    <span
      class="viaText"
    >
      
      via
    
    </span>
  </span>
   
  <fa-stub
    class="icon"
    icon="[object Object]"
    size="xs"
  />
   
  <fa-stub
    class="icon"
    icon="[object Object]"
    size="sm"
  />
   
  <fa-stub
    class="icon"
    icon="[object Object]"
    size="xs"
  />
</nuxt-link-stub>
`;
